### DISCUSSION TITLE
can the files be zipped?

### DISCUSSION THREAD

**Discussion by nmayle** [CMSgov/price-transparency-guide#59] [Q&A] [answered]
Can the files once created in the correct format be zipped and put on the website as .zip files?  If so, can the zip archive contain all three files?

**Comment by shaselton-usds** [comment, marked as answer]
There are no restrictions in zipping the files.

**Reply by ghost** [reply]
Thank you! I will be passing this along to our infrastructure team and they will be very glad to hear this.

**Reply by nmayle** [reply]
Are there any restrictions on what types of techniques and formats can be used for compressing the files?
Zip is one of many, although perhaps the most common e.g. gzip, 7z

**Reply by balbarka** [reply]
I recommend using gzip, *.json.gz for file generation. With proper end line formatting within element arrays, you are able to directly read the *.json.gz files in spark / hive using a json serde like org.openx.data.jsonserde.JsonSerDe. This will be useful in validation and consumption ingest.

**Comment by BrianMI** [comment]
Follow up question to this thread... If we were to zip the JSON, does the zip file need to follow the same naming convention as the JSON?  For example, could the zip file be called issuer-abc_healthcare-100_in-network-rates.zip and when unzipped the file would be 2020-01-05_issuer-abc_healthcare-100_in-network-rates.json?

Trying to solve the issue of the file sizes as well as having a static filename to link to from the webpage.

**Reply by balbarka** [reply]
I recommend using gzip which by default converts *.json.gz -> *.json when using cli gzip -d <file>. Thus, we have the same file convention for the compressed as the uncompressed with .gz as the only difference. If we happen to want to put multiple gzip files into a single artifact, we would then have to tar the file, but I haven't run into this requirement yet.

**Comment by nmayle** [comment]
The question about the name of the zip files and whether all the files could be zipped into one archive is a bit important since by having a standard name for the files, they may be easier to find via searches etc.

**Reply by shaselton-usds** [reply]
At this point, there aren't any naming restrictions on the zipped archive. Your point RE: searchability is definitely valid. To be helpful on this front, you can definitely follow the naming schema for the files for the zipped archive. 

If there is a plan to combine multiple files together in the zipped archive, naming gets tricky in the context of web discoverability. SEO best practices can definitely help out but that goes beyond and a bit too prescriptive than what the final rules lay out.

This is actually a bit (slightly) related to the discussion happening over at: https://github.com/CMSgov/price-transparency-guide/discussions/99. How to preserve the naming for a single plan when a file may contain multiple plans. This could be potentially solved by have symlinks with the unique names pointing to the single file to be downloaded that includes the plan's data, but I fear that we are solutioning at this point. 

cc @BobSyracuse @BrianMI

**Reply by nmayle** [reply]
Looking at the language of the rule and the discussion in the Federal Register, it seems that the requirement is to post three files:

> . The final rules also require plans and issuers to disclose in-network provider negotiated rates, historical out- of-network allowed amounts, and drug pricing information through three machine-readable files posted on an internet website, thereby allowing the public to have access to health coverage information that can be used to understand health care pricing and potentially dampen the rise in health care spending. 

So, zipping them into one file would not fit this description.

**Reply by shaselton-usds** [reply]
Agreed, we are not combining multiple files into a single file -- this question is more about compressing multiple files into a single archive.

**Reply by nmayle** [reply]
If it is a single archive file on the website, does that still meet the requirement of "three machine-readable files posted on an internet website"?  If a tool-builder is interested in just the pharma data or one of the other types, then they would have to download the whole zip and deal with all three files when they only want one.
